From 96a32d952d4014df0831cfb1f3a1fd8e4bedfb6d Mon Sep 17 00:00:00 2001
From: Meriipu <Meriipu@users.noreply.github.com>
Date: Fri, 17 Dec 2021 16:02:05 +0100
Subject: [PATCH 07/10] Removed from contextmenu:

* email-image
* save-snapshot,
* send-to-device
* inspect accessibility
* set background
* pocket trash

I do not use them. Countless times have I clicked email image, and
occasionally I have accidentally saved a frame instead of saving the
video itself.
---
 browser/base/content/browser-context.inc | 61 ++++--------------------
 1 file changed, 10 insertions(+), 51 deletions(-)

diff --git a/browser/base/content/browser-context.inc b/browser/base/content/browser-context.inc
index 03799874d5b2..a76d5116488f 100644
--- a/browser/base/content/browser-context.inc
+++ b/browser/base/content/browser-context.inc
@@ -101,24 +101,12 @@
       <menuitem id="context-savelink"
                 data-l10n-id="main-context-menu-save-link"
                 oncommand="gContextMenu.saveLink();"/>
-      <menuitem id="context-savelinktopocket"
-                data-l10n-id="main-context-menu-save-link-to-pocket"
-                oncommand= "Pocket.savePage(gContextMenu.browser, gContextMenu.linkURL);"/>
       <menuitem id="context-copyemail"
                 data-l10n-id="main-context-menu-copy-email"
                 oncommand="gContextMenu.copyEmail();"/>
       <menuitem id="context-copylink"
                 data-l10n-id="main-context-menu-copy-link-simple"
                 oncommand="gContextMenu.copyLink();"/>
-      <menu id="context-sendlinktodevice"
-            class="sync-ui-item"
-            data-l10n-id="main-context-menu-link-send-to-device"
-            hidden="true">
-            <menupopup id="context-sendlinktodevice-popup"
-                  onpopupshowing="gSync.populateSendTabToDevicesMenu(event.target, gContextMenu.linkURL, gContextMenu.linkTextStr);"/>
-      </menu>
-      <menuseparator id="context-sep-sendlinktodevice" class="sync-ui-item"
-                hidden="true"/>
       <menuseparator id="context-sep-copylink"/>
       <menuitem id="context-media-play"
                 data-l10n-id="main-context-menu-media-play"
@@ -193,18 +181,6 @@
       <menuitem id="context-viewimage"
                 data-l10n-id="main-context-menu-image-view-new-tab"
                 oncommand="gContextMenu.viewMedia(event);"/>
-      <menuitem id="context-saveimage"
-                data-l10n-id="main-context-menu-image-save-as"
-                oncommand="gContextMenu.saveMedia();"/>
-      <menuitem id="context-video-saveimage"
-                data-l10n-id="main-context-menu-video-take-snapshot"
-                oncommand="gContextMenu.saveVideoFrameAsImage();"/>
-      <menuitem id="context-savevideo"
-                data-l10n-id="main-context-menu-video-save-as"
-                oncommand="gContextMenu.saveMedia();"/>
-      <menuitem id="context-saveaudio"
-                data-l10n-id="main-context-menu-audio-save-as"
-                oncommand="gContextMenu.saveMedia();"/>
 #ifdef CONTEXT_COPY_IMAGE_CONTENTS
       <menuitem id="context-copyimage-contents"
                 data-l10n-id="main-context-menu-image-copy"
@@ -219,15 +195,16 @@
       <menuitem id="context-copyaudiourl"
                 data-l10n-id="main-context-menu-audio-copy-link"
                 oncommand="gContextMenu.copyMediaLocation();"/>
-      <menuitem id="context-sendimage"
-                data-l10n-id="main-context-menu-image-email"
-                oncommand="gContextMenu.sendMedia();"/>
-      <menuitem id="context-sendvideo"
-                data-l10n-id="main-context-menu-video-email"
-                oncommand="gContextMenu.sendMedia();"/>
-      <menuitem id="context-sendaudio"
-                data-l10n-id="main-context-menu-audio-email"
-                oncommand="gContextMenu.sendMedia();"/>
+      <menuseparator id="context-sep-copyimage"/>
+      <menuitem id="context-saveimage"
+                data-l10n-id="main-context-menu-image-save-as"
+                oncommand="gContextMenu.saveMedia();"/>
+      <menuitem id="context-savevideo"
+                data-l10n-id="main-context-menu-video-save-as"
+                oncommand="gContextMenu.saveMedia();"/>
+      <menuitem id="context-saveaudio"
+                data-l10n-id="main-context-menu-audio-save-as"
+                oncommand="gContextMenu.saveMedia();"/>
       <menuitem id="context-viewimageinfo"
                 hidden="true"
                 data-l10n-id="main-context-menu-image-info"
@@ -235,10 +212,6 @@
       <menuitem id="context-viewimagedesc"
                 data-l10n-id="main-context-menu-image-desc"
                 oncommand="gContextMenu.viewImageDesc(event);"/>
-      <menuseparator id="context-sep-setbackground"/>
-      <menuitem id="context-setDesktopBackground"
-                data-l10n-id="main-context-menu-image-set-image-as-background"
-                oncommand="gContextMenu.setDesktopBackground();"/>
       <menuitem id="context-ctp-play"
                 data-l10n-id="main-context-menu-plugin-play"
                 oncommand="gContextMenu.playPlugin();"/>
@@ -254,16 +227,6 @@
       <menuitem id="context-savepage"
                 data-l10n-id="main-context-menu-page-save"
                 oncommand="gContextMenu.savePageAs();"/>
-      <menuitem id="context-pocket"
-                data-l10n-id="main-context-menu-save-to-pocket"
-                oncommand="Pocket.savePage(gContextMenu.browser, gContextMenu.browser.currentURI.spec, gContextMenu.browser.contentTitle);"/>
-      <menu id="context-sendpagetodevice"
-            class="sync-ui-item"
-            data-l10n-id="main-context-menu-send-to-device"
-            hidden="true">
-        <menupopup id="context-sendpagetodevice-popup"
-                   onpopupshowing="(() => { gSync.populateSendTabToDevicesMenu(event.target, gBrowser.currentURI.spec, gBrowser.contentTitle); })()"/>
-      </menu>
       <menu id="fill-login" hidden="true">
         <menupopup id="fill-login-popup" />
       </menu>
@@ -394,10 +357,6 @@
       <menuitem id="context-viewsource"
                 data-l10n-id="main-context-menu-view-page-source"
                 oncommand="BrowserViewSource(gContextMenu.browser);"/>
-      <menuitem id="context-inspect-a11y"
-                hidden="true"
-                data-l10n-id="main-context-menu-inspect-a11y-properties"
-                oncommand="gContextMenu.inspectA11Y();"/>
       <menuitem id="context-inspect"
                 hidden="true"
                 data-l10n-id="main-context-menu-inspect"
-- 
2.34.1

